tests/algorithms/test_cg.py

```diff
@@ -0,0 +1,87 @@
+"""Tests for the conjugate gradient method."""
+
+# Copyright 2024 Physikalisch-Technische Bundesanstalt
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#       http://www.apache.org/licenses/LICENSE-2.0
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import pytest
+import torch
+
+from mrpro.algorithms import cg
+from mrpro.operators import LinearOperator
+from tests import RandomGenerator
+
+
+def create_self_adjoint_op(N, complex_valued_flag):
+    """Create self-adjoint operator H and wrap as LinearOperator."""
+    random_generator = RandomGenerator(seed=0)
+    N = 32
+
+    # generate random matrices (avoid zeros by using ab offset of 1.)
+    if complex_valued_flag:
+        A = 1 + random_generator.complex64_tensor(size=(N, N))
+        H = A.conj().T @ A
+    else:
+        A = 1 + random_generator.float32_tensor(size=(N, N))
+        H = A.T @ A
+
+    class MatMult(LinearOperator):
```

Comment:
Just wondering, maybe we could put this into the main of mrpro as a MatrixOperator ?
Thi might be useful for other stuff as well